Addition of resync api for migration and fixing ReplicationMigrationItem examples

anmolbhatia289 opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

anmolbhatia289 committed:
adding resync api changes and fixing examples

anmolbhatia289 committed:
reverting addition of vaultLocation and deletion of read-only property

anmolbhatia289 committed:
Merge pull request #12 from AsrOneSdk/anbhat/masterCurrent

Adding resync api changes and fixing examples

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=7863) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=7863) before we can accept your contribution.<br/><br/>:x: anmolbhatia289 [sign now](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=7863)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure/azure-rest-api-specs?pullRequest=7863) it.</sub>

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-go

anmolbhatia289 commented:
**Changes**

1. Added Path: /Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationMigrationItems/{migrationItemName}/resync
 For enabling manual resync.

2. Added Model: ResyncInput, ResyncInputProperties, ResyncProviderSpecificInput, VMwareCbtResyncInput

3. Modified Model: EventProperties
   - Added property affectedObjectCorrelationId


4. Modified Model: EventQueryParameter
   - Added property affectedObjectCorrelationId

5. Modified Model: MigrationItemProperties
   - Removed property recoveryServicesProviderId
   - Added property lastTestMigrationTime
   - Added property lastTestMigrationStatus
   - Modified property health eventCorrelationId
   - Modified property allowedOperations

6. Modified Model: TestMigrateCleanupInputProperties
   - Added property maxLength

7. Modified Model: VMwareCbtDiskInput
   - Modified property, input of isOSDisk is of type boolean

8. Modified model
   - Added property performAutoResync

9. Added model VMwareCbtEventDetails

10. Modified model VMwareCbtMigrateInput
    - Modified property, input of performShutDown is of type boolean

11. Modified model VMwareCbtMigrationDetails
    - Added property lastRecoveryPointId, initialSeedingProgressPercentage, migrationProgressPercentage, resyncProgressPercentage, resyncState, performAutoResync

12. Modified model VMwareCbtNicDetails
    - Modified property, input of isPrimaryNic is of type boolean
    - Modified property, input of isSelectedForMigration is of type boolean

13. Modified model VMwareCbtNicInput
    -Modified property, input of isPrimaryNic is of type boolean
    - Modified property, input of isSelectedForMigration is of type boolean

14. Modified model VMwareCbtProtectedDiskDetails
    - Modified property, input of isOSDisk is of type boolean

15. Modified model VMwareCbtUpdateMigrationItemInput
    - Added property performAutoResync

16. Modified model VMwareV2FabricCreationInput
    - Added property physicalSiteId


17. Modified model VMwareV2FabricCreationInput
    - Added property physicalSiteId
    - Added property serviceResourceId

18. Modified examples, according to these change and addition of models. 

Current semantic errors : 
    - instanceType DISCRIMINATOR_NOT_REQUIRED for few models

Current model validation errors: 
    - READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST for instanceType property for models FabricSpecificCreationInput and SwitchProtection
    - OBJECT_MISSING_REQUIRED_PROPERTY for vaultLocation missing in examples, this property is obselete and we are planning a separate PR for this. 

semantic errors and model validation error READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST, are both concerned with instanceType property. This will be fixed in a separate PR.

anmolbhatia289 committed:
changing added boolean properties to string

anmolbhatia289 committed:
Merge pull request #13 from AsrOneSdk/anbhat/masterCurrent

Changing boolean properties to string

akning-ms commented:
@anmolbhatia289 Is your PR ready for review? pls fix few validation error first

anmolbhatia289 commented:
Yes, it is ready for review. There are validation errors in already checked in specs as well. We are planning to fix current as well as already present errors with a separate PR.

KrisBash commented on specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json:
```diff
@@ -8781,6 +8858,10 @@
           "description": "The friendly name of the source of the event on which it is raised (for example, VM, VMM etc).",
           "type": "string"
         },
+        "affectedObjectCorrelationId": {
```
Is this read-only?

KrisBash commented on specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json:
```diff
@@ -8847,6 +8928,10 @@
           "description": "The affected object name of the events to be queried.",
           "type": "string"
         },
+        "affectedObjectCorrelationId": {
```
read-only?

KrisBash commented on specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json:
```diff
@@ -12332,11 +12417,6 @@
           "type": "string",
           "readOnly": true
         },
-        "recoveryServicesProviderId": {
```
Removal of a property is considered a breaking change. Please do this in a new version.

anmolbhatia289 replied:
This is no longer required to be exposed to the customer. This is a read-only property, and removing it should not be a breaking change.

KrisBash commented on specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-01-10/service.json:
```diff
@@ -12421,6 +12523,10 @@
           "description": "The current job details.",
           "readOnly": true
         },
+        "eventCorrelationId": {
```
Is this read-only?

KrisBash reviewed (changes requested):
Please see comments

anmolbhatia289 committed:
fixing comments and 2018_07_10 migration examples

anmolbhatia289 committed:
fixing readOnly attribute

anmolbhatia289 committed:
Merge pull request #14 from AsrOneSdk/anbhat/masterCurrent

Adding readOnly attributes to fix comments and fixing 2018_07_10 migration examples

AutorestCI commented:
You don't have permission to trigger SDK Automation.
Please add yourself to ***Azure*** group from opensource portal if you are MSFT employee,
or please ask reviewer to add comment *** /openapibot sdkautomation ***.
Please ask tih@microsoft.com (or NullMDR in github) for additional help.

azuresdkci commented:
Can one of the admins verify this patch?

akning-ms commented:
> Yes, it is ready for review. There are validation errors in already checked in specs as well. We are planning to fix current as well as already present errors with a separate PR.

Just curious, why not fix in this same PR?

akning-ms commented:
removed Lable ARMChangesRequested  for ARM confirm and signoff

anmolbhatia289 commented:
> > Yes, it is ready for review. There are validation errors in already checked in specs as well. We are planning to fix current as well as already present errors with a separate PR.
> 
> Just curious, why not fix in this same PR?

Model validation failures and semantic failures have nothing to do with the new change of adding resync API. They have already been fixed by now in this PR : https://github.com/Azure/azure-rest-api-specs/pull/7982

Commit:
Solving model validation errors - removing readOnly tags from request properties, correcting typo in example, removing non-mandatory property marked as mandatory.

Commit:
Model validation failure fixes for 2018-01-10 version.

punit1396 commented:
Have added changes to fix the modelValidation issues. The semantic issues have been fixed in a separate PR. Since all the required checks are now passing, @akning-ms, could we merge and close this PR?
